### PR TITLE
Add docs for installing nginx-agent from repository

### PR DESCRIPTION
**Pull request by Dean-Coakley**
### Proposed changes

* Add install instructions for installing from NGINX OSS repository
* Add install instructions for installing from NGINX Plus repository
* Fix typos
* Move logs and systemd info from install doc to getting-started doc
* Add upgrade doc

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [x] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: hugo/content/getting-started.md [hunk 1]
```diff
@@ -139,3 +139,23 @@ Open a web browser to view the mock control plane at [http://localhost:54790](ht
 - **metrics** - shows a buffer of metrics sent to the management plane (similar to what will be sent back in the REST API)
 
 For more NGINX Agent use cases, refer to the [NGINX Agent SDK examples](https://github.com/nginx/agent/tree/main/sdk/examples).
+
+## Start and Enable Start on Boot
+
+To start the NGINX Agent on `systemd` systems, run the following command:
+
+```
```
Comment (Jcahilltorre) [inline hugo/content/getting-started.md:147]:
```suggestion
```bash
```

File: hugo/content/installation-plus.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+doctypes: ["task"]
+---
+
+## Install NGINX Plus
+
+NGINX Agent interfaces directly with an NGINX server process installed on the same system. If you don't have it already, follow these steps to install [NGINX Plus](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-plus/). Once installed, ensure the NGINX instance is running.
+
+## Install NGINX Agent from NGINX Plus Repository
+
+First ensure that you are running a supported distribution and architecture: [technical specifications](../technical-specifications/)
```
Comment (Jcahilltorre) [inline hugo/content/installation-plus.md:19]:
```suggestion
First, ensure that you are running a supported distribution and architecture: [technical specifications](../technical-specifications/)
```

File: hugo/content/installation-plus.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    ```
+    sudo zypper install curl ca-certificates gpg2 gawk
+    ```
+
+  - To set up the zypper repository for `nginx-agent` packages, run the following command:
+    ```
+    sudo zypper addrepo --refresh --check \
+        'https://pkgs.nginx.com/nginx-agent/sles/$releasever_major?ssl_clientcert=/etc/ssl/nginx/nginx-repo-bundle.crt&ssl_verify=peer' nginx-agent
+    ```
+  - Next, import an official nginx signing key so `zypper`/`rpm` could verify the packages authenticity. Fetch the key:
```
Comment (Jcahilltorre) [inline hugo/content/installation-plus.md:161]:
```suggestion
  - Next, import an official NGINX signing key so `zypper`/`rpm` can verify the package's authenticity. Fetch the key:
```

File: hugo/content/installation-plus.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    ```
+    printf "%s%s%s%s\n" \
+        "@nginx-agent " \
+        "https://pkgs.nginx.com/nginx-agent/alpine/v" \
+        `grep -o -E '^[0-9]+\.[0-9]+' /etc/alpine-release` \
+        "/main" \
+        | sudo tee -a /etc/apk/repositories
+    ```
+
+  - Next, import an official nginx signing key so apk could verify the packages authenticity. Fetch the key:
```
Comment (Jcahilltorre) [inline hugo/content/installation-plus.md:214]:
```suggestion
  - Next, import an official NGINX signing key so apk can verify the package's authenticity. Fetch the key:
```

File: hugo/content/overview.md [hunk 5]
```diff
@@ -5,7 +5,7 @@ description: "Learn about NGINX Agent and how it works."
 weight: 100
 toc: true
 tags: [ "docs" ]
-docs: "DOCS-1091"
+docs: "DOCS-1093"
```
Comment (Jcahilltorre) [inline hugo/content/overview.md:8]:
```suggestion
docs: "DOCS-1091"
```

File: hugo/content/technical-specifications.md [hunk 6]
```diff
@@ -2,10 +2,10 @@
 title: "Technical Specifications"
 draft: false
 description: "This document describes the requirements for NGINX Agent."
-weight: 500
+weight: 700
 toc: true
 tags: [ "docs" ]
-docs: "DOCS-1092"
+docs: "DOCS-1094"
```
Comment (Jcahilltorre) [inline hugo/content/technical-specifications.md:8]:
```suggestion
docs: "DOCS-1092"
```

**Review by Jcahilltorre — CHANGES_REQUESTED** [review]
comments + suggestions

File: hugo/content/installation-oss.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+draft: false
+description: "Learn how to install NGINX Agent from OSS Repository."
+weight: 300
+toc: true
+tags: [ "docs" ]
+docs: "DOCS-1216"
+categories: ["configuration"]
+doctypes: ["task"]
+---
+
```
Comment (craigell) [inline hugo/content/installation-oss.md:12]: We should probably include a pre-requisites section that mentions that NGINX OSS is installed and a supported operating system with a link to our supported OS's. Example can be found here https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-plus/#prerequisites

File: hugo/content/installation-oss.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+- [RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux](#rhel-centos-rocky-linux-almalinux-and-oracle-linux)
+- [Ubuntu](#ubuntu)
+- [Debian](#debian)
+- [SLES](#sles)
+- [Alpine Linux](#alpine-linux)
+- [Amazon Linux](#amazon-linux)
+
+### RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux
+  - Install the prerequisites:
```
Comment (craigell) [inline hugo/content/installation-oss.md:31]: Should include numbered steps to replace of bullet points.

File: hugo/content/installation-oss.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+    module_hotfixes=true
+    ```
+
+  - To install nginx, run the following command:
+
+    ```bash
+    sudo yum install nginx-agent
+    ```
+
+  - When prompted to accept the GPG key, verify that the fingerprint matches `573B FD6B 3D8F BC64 1079 A6AB ABF5 BD82 7BD9 BF62`, and if so, accept it.
```
Comment (craigell) [inline hugo/content/installation-oss.md:248]: Think we should include a upgrade section for the agent. https://docs.nginx.com/nginx-management-suite/admin-guides/installation/upgrade-guide/#upgrade-nginx-agent

File: hugo/content/installation-plus.md [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+    ```
+
+1. Install the prerequisites:
+
+    ```shell
+    sudo yum install yum-utils procps
+    ```
+
+1. To set up the yum repository, create the file named `/etc/yum.repos.d/nginx-agent.repo` with the following contents:
+
```
Comment (craigell) [inline hugo/content/installation-plus.md:56]: Need to add command to this step create repo file in that directory. See step 2 from existing doc https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/#installing-a-prebuilt-rhelcentosoracle-linuxalmalinuxrocky-linux-package-from-the-official-nginx-repository
``` shell
 sudo vi /etc/yum.repos.d/nginx-agent.repo

```

File: hugo/content/installation-plus.md
Comment (craigell) [inline hugo/content/installation-plus.md:1]: Need to add FreeBSD instructions https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-plus/#installing-nginx-plus-on-freebsd

File: hugo/content/installation-oss.md
Comment (craigell) [inline hugo/content/installation-oss.md:1]: Need to add FreeBSD instructions

File: hugo/content/installation-plus.md [hunk 11]
```diff
@@ -0,0 +1,453 @@
+---
+title: "Install NGINX Agent From NGINX Plus Repository"
```
Comment (craigell) [inline hugo/content/installation-plus.md:2]: Title to change to "Installation from NGINX Plus Repository" to be in line with the OSS title.

File: hugo/content/installation-oss.md [hunk 12]
```diff
@@ -0,0 +1,349 @@
...
+
+- NGINX installed. Once installed, ensure it is running. If you don't have it installed already, follow these steps to install [NGINX](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/)
+- A [supported operating system and architecture](../technical-specifications/#supported-distributions)
+- `root` privilege
+
+## Configure NGINX OSS Repository for installing NGINX Agent
+
+Before you install NGINX Agent for the first time on your system, you need to set up the `nginx-agent` packages repository. Afterward, you can install and update NGINX Agent from the repository.
+
+- [Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux](#installing-nginx-agent-on-rhel-centos-rocky-linux-almalinux-and-oracle-linux)
```
Comment (Jcahilltorre) [inline hugo/content/installation-oss.md:24]:
```suggestion
- [Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux, and Oracle Linux](#installing-nginx-agent-on-rhel-centos-rocky-linux-almalinux-and-oracle-linux)
```

File: hugo/content/installation-oss.md [hunk 13]
```diff
@@ -0,0 +1,349 @@
...
+- [Installing NGINX Agent on Ubuntu](#installing-nginx-agent-on-ubuntu)
+- [Installing NGINX Agent on Debian](#installing-nginx-agent-on-debian)
+- [Installing NGINX Agent on SLES](#installing-nginx-agent-on-sles)
+- [Installing NGINX Agent on Alpine Linux](#installing-nginx-agent-on-alpine-linux)
+- [Installing NGINX Agent on Amazon Linux](#installing-nginx-agent-on-amazon-linux)
+- [Installing NGINX Agent on FreeBSD](#installing-nginx-agent-on-freebsd)
+
+### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux
```
Comment (Jcahilltorre) [inline hugo/content/installation-oss.md:32]:
```suggestion
### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux, and Oracle Linux
```

File: hugo/content/installation-plus.md [hunk 14]
```diff
@@ -0,0 +1,453 @@
...
+- A [supported operating system and architecture](../technical-specifications/#supported-distributions)
+- `root` privilege
+- Your credentials to the MyF5 Customer Portal, provided by email from F5, Inc.
+- Your NGINX Plus certificate and public key (`nginx-repo.crt` and `nginx-repo.key` files), provided by email from F5, Inc.
+
+## Configure NGINX Plus Repository for installing NGINX Agent
+
+Before you install NGINX Agent for the first time on your system, you need to set up the `nginx-agent` packages repository. Afterward, you can install and update NGINX Agent from the repository.
+
+- [Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux](#installing-nginx-agent-on-rhel-centos-rocky-linux-almalinux-and-oracle-linux)
```
Comment (Jcahilltorre) [inline hugo/content/installation-plus.md:27]:
```suggestion
- [Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux, and Oracle Linux](#installing-nginx-agent-on-rhel-centos-rocky-linux-almalinux-and-oracle-linux)
```

File: hugo/content/installation-plus.md [hunk 15]
```diff
@@ -0,0 +1,453 @@
...
+- [Installing NGINX Agent on Ubuntu](#installing-nginx-agent-on-ubuntu)
+- [Installing NGINX Agent on Debian](#installing-nginx-agent-on-debian)
+- [Installing NGINX Agent on SLES](#installing-nginx-agent-on-sles)
+- [Installing NGINX Agent on Alpine Linux](#installing-nginx-agent-on-alpine-linux)
+- [Installing NGINX Agent on Amazon Linux](#installing-nginx-agent-on-amazon-linux)
+- [Installing NGINX Agent on FreeBSD](#installing-nginx-agent-on-freebsd)
+
+### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux
```
Comment (Jcahilltorre) [inline hugo/content/installation-plus.md:35]:
```suggestion
### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux, and Oracle Linux
```

**Review by craigell — APPROVED** [review]

**Pull request merged**